File: js-archive/lexicographically-smallest-palindrome.js
/**
 * @param {string} s
 * @return {string}
 */
var makeSmallestPalindrome = function(s) {
  let ans = Array(s.length).fill('')
  let [l, r] = [0, s.length - 1]

  while (l <= r) {
    if (s[l].charCodeAt(0) > s[r].charCodeAt(0)) {
      ans[l] = s[r]
      ans[r] = s[r]
    } else {
      ans[l] = s[l]
      ans[r] = s[l]
    }

    l++
    r--
  }

  return ans.join('')
};

File: lexicographically-smallest-palindrome.go
func makeSmallestPalindrome(s string) string {
    r:=[]rune(s)

    for i:=0; i <len(r) / 2; i++ {
        if int(r[i]) > int(r[len(r) - 1 - i]) {
            r[i] = r[len(r) - 1 - i]
        } else {
            r[len(r) - 1 - i] = r[i]
        }
    }

    return string(r)
}